### PR TITLE
feat(export): manager VMWare V6 config export

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR intends to generate export configuration file for VMWare V6 ACC Configuration.
The file is generate like this:
```json
{
    "vsphere_server": [
        {
            "name": "my-vcenter A",
            "url": "https://127.0.0.2/sdk",
            "username": "my-username A",
            "password": "my-password A"
        },
        {
            "name": "my-vcenter B",
            "url": "https://127.0.0.3/sdk",
            "username": "my-username B",
            "password": "my-password B"
        }
    ],
    "port": 442
}
```

The VMWareV6 configuration files are created for each pollers in:
`<configured_centre_cache_dir>/config/vmware/<pollerid>/centreon_vmware.json`

**Fixes** # MON-138541

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [x] master

<h2> How this pull request can be tested ? </h2>
- Create an ACC VMWare V6 linked to multiple pollers.
- Export your pollers configuration
- Each configuration files should have been created

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/AdditionalConnectorConfiguration/Domain/Model/VmWareV6/VSphereServer.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+namespace Core\AdditionalConnectorConfiguration\Domain\Model\VmWareV6;
+
+class VSphereServer
+{
+    public function __construct(
+        private readonly string $name,
+        private readonly string $url,
+        private readonly string $username,
+        private readonly string $password
```
Comment (callapa) [inline centreon/src/Core/AdditionalConnectorConfiguration/Domain/Model/VmWareV6/VSphereServer.php:32]: No property length control?

File: centreon/src/Core/AdditionalConnectorConfiguration/Domain/Model/VmWareV6/VmWareConfig.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    public function __construct(private readonly array $vSphereServers, private readonly int $port)
+    {
+    }
+
+    /**
+     * @return VSphereServer[]
+     */
+    public function getVSphereServers(): array
+    {
+        return $this->vSphereServers;
```
Comment (callapa) [inline centreon/src/Core/AdditionalConnectorConfiguration/Domain/Model/VmWareV6/VmWareConfig.php:41]: There is no guarantee that the property returns a vSphereServers array.

File: centreon/src/Core/AdditionalConnectorConfiguration/Domain/Model/VmWareV6/VmWareConfig.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+namespace Core\AdditionalConnectorConfiguration\Domain\Model\VmWareV6;
+
+class VmWareConfig
+{
+    /**
+     * @param VSphereServer[] $vSphereServers
+     * @param int $port
+     */
+    public function __construct(private readonly array $vSphereServers, private readonly int $port)
```
Comment (callapa) [inline centreon/src/Core/AdditionalConnectorConfiguration/Domain/Model/VmWareV6/VmWareConfig.php:32]: No property control?

File: centreon/www/class/config-generate/AdditionalConnectorVmWareV6.class.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+ */
+
+use App\Kernel;
+use Core\AdditionalConnectorConfiguration\Application\Repository\ReadAccRepositoryInterface;
+use Core\AdditionalConnectorConfiguration\Domain\Model\Type;
+use Core\AdditionalConnectorConfiguration\Domain\Model\VmWareV6\{VmWareConfig, VSphereServer};
+
+class AdditionalConnectorVmWareV6 extends AbstractObjectJSON
+{
+    public function __construct(\Pimple\Container $dependencyInjector)
```
Comment (callapa) [inline centreon/www/class/config-generate/AdditionalConnectorVmWareV6.class.php:29]: It's not good practice to inject a Container directly into a constructor. It's better to select the necessary classes

**Review by callapa — CHANGES_REQUESTED** [review]

File: centreon/www/class/config-generate/AdditionalConnectorVmWareV6.class.php (hunk 4 above)
Comment (jeremyjaouen) [reply centreon/www/class/config-generate/AdditionalConnectorVmWareV6.class.php:29]: The parent needs the Container :/

File: centreon/src/Core/AdditionalConnectorConfiguration/Domain/Model/VmWareV6/VSphereServer.php (hunk 1 above)
Comment (jeremyjaouen) [reply centreon/src/Core/AdditionalConnectorConfiguration/Domain/Model/VmWareV6/VSphereServer.php:32]: There is no really needs, as when we create an ACC everything is store as a json 
```json
{
    "port": 442,
    "vcenters": [
        {
            "name": "my-vcenter A",
            "url": "https://127.0.0.1/sdk",
            "username": "<encrypted_username>",
            "password": "<encrypted_password>"
        },
        {
            "name": "my-vcenter B",
            "url": "https://127.0.0.2/sdk",
            "username": "<encrypted_username>",
            "password": "<encrypted_password>"
        },
        {
            "name": "my-vcenter C",
            "url": "https://127.0.0.2/sdk",
            "username": "<encrypted_username>",
            "password": "<encrypted_password>"
        }
    ]
}
```

File: centreon/src/Core/AdditionalConnectorConfiguration/Domain/Model/VmWareV6/VmWareConfig.php (hunk 2 above)
Comment (jeremyjaouen) [reply centreon/src/Core/AdditionalConnectorConfiguration/Domain/Model/VmWareV6/VmWareConfig.php:41]: I'll check in constructor that every item in the array is an instance of VSphereServer

File: centreon/src/Core/AdditionalConnectorConfiguration/Domain/Model/VmWareV6/VmWareConfig.php (hunk 3 above)
Comment (callapa) [reply centreon/src/Core/AdditionalConnectorConfiguration/Domain/Model/VmWareV6/VmWareConfig.php:37]: Perfect.
Other solution, you can create a new method `AddServer(VSphereServer $server): void` and use it in your iteration.

**Review by callapa — DISMISSED** [review]

**Review by a-launois — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**